temoa/data_processing/database_util.py

```
@@ -41,361 +47,284 @@ def __init__(self, database_path, scenario=None):
 
-    def get_capacity_for_tech_and_period(self, tech=None, period=None, region=None):
-        if self.cur is None:
-            raise ValueError('Invalid Operation For dat file')
-        if self.scenario is None or self.scenario == '':
-            raise ValueError('For Output related queries, please set a scenario first')
+    def get_capacity_for_tech_and_period(
+        self, tech: str | None = None, period: int | None = None, region: str | None = None
+    ) -> pd.DataFrame | pd.Series:
+        """Retrieves capacity data, aggregated by technology."""
+        if not self.scenario:
+            raise ValueError('A scenario must be set for output-related queries')
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)